File: CodingChallengeSolutionsC++/2145. Count the Hidden Sequences/2145.CounttheHiddenSequences.cs
namespace CodingChallengeSolutions.CounttheHiddenSequences
{
    public class CounttheHiddenSequences
    {
        public int NumberOfArrays(int[] differences, int lower, int upper)
        {
            long preSum = 0;
            long minVal = 0;
            long maxVal = 0;
            for (int x = 0; x < differences.Count(); ++x)
            {
                preSum += differences[x];
                minVal = Math.Min(minVal, preSum);
                maxVal = Math.Max(maxVal, preSum);
            }

            return Math.Max(0, (upper - lower + 1) - (int)(maxVal - minVal));
        }
    }
}

File: CodingChallengeSolutionsC++/2145. Count the Hidden Sequences/2145.CounttheHiddenSequences.js
var numberOfArrays = function (differences, lower, upper) {
    preSum = 0;
    minVal = 0;
    maxVal = 0;
    for (x = 0; x < differences.length; ++x) {
        preSum += differences[x];
        minVal = Math.min(minVal, preSum);
        maxVal = Math.max(maxVal, preSum);
    }

    return Math.max(0, (upper - lower + 1) - (maxVal - minVal));
};